src/Grapevine.Tests/Server/PublicFolderFacts.cs

```diff
@@ -18,133 +19,209 @@ private static string GenerateUniqueString()
+                var watcher = new FileSystemWatcher
+                {
+                    Path = folder.FolderPath,
+                    Filter = "*.jpg",
+                    EnableRaisingEvents = true,
+                    IncludeSubdirectories = true,
+                    NotifyFilter = NotifyFilters.FileName
+                };
 
-                var folderpath = Directory.CreateDirectory(Path.Combine(root.FolderPath, folder)).FullName;
-                if (!Directory.Exists(folderpath)) throw new Exception("Folder to test did not get created");
+                folder.Watcher.Filter.ShouldNotBe(watcher.Filter);
 
-                root.FilePathGetter($"/{folder}").ShouldBeNull();
+                folder.Watcher = watcher;
 
-                CleanUp(folderpath);
+                folder.Watcher.Filter.ShouldBe(watcher.Filter);
             }
 
             [Fact]
-            public void ReturnsExistingFilePath()
+            public void DoesNotUpdateToNull()
             {
-                var root = new PublicFolder();
-                var file = GenerateUniqueString();
+                var folder = new PublicFolder();
+                var watcher = folder.Watcher;
 
-                var filepath = Path.Combine(root.FolderPath, file);
-                using (var sw = File.CreateText(filepath)) { sw.WriteLine("Hello"); }
+                folder.Watcher = null;
 
-                root.FilePathGetter(file).ShouldBe(filepath);
-                root.FilePathGetter($"/{file}").ShouldBe(filepath);
-
-                File.Delete(filepath);
+                folder.Watcher.ShouldNotBeNull();
+                folder.Watcher.Equals(watcher).ShouldBeTrue();
             }
 
             [Fact]
-            public void ReturnsExistingDefaultFilePath()
+            public void DoesNotDisposeWhenSetToSameValue()
             {
-                var root = new PublicFolder();
-                var folder = GenerateUniqueString();
+                var watcher = Substitute.For<FileSystemWatcher>();
+                var folder = new PublicFolder {Watcher =  watcher};
 
-                var folderpath = Directory.CreateDirectory(Path.Combine(root.FolderPath, folder)).FullName;
-                if (!Directory.Exists(folderpath)) throw new Exception("Folder to test did not get created");
+                folder.Watcher = watcher;
+
+                watcher.DidNotReceive().Dispose();
+            }
+        }
+
+        public class FolderPathProperty
+        {
+            [Fact]
+            public void CreatesFolderIfNotExists()
+            {
+                var path = Path.Combine(Directory.GetCurrentDirectory(), GenerateUniqueString());
+                Directory.Exists(path).ShouldBeFalse();
 
-                var filepath = Path.Combine(folderpath, root.DefaultFileName);
-                using (var sw = File.CreateText(filepath)) { sw.WriteLine("Hello"); }
+                var folder = new PublicFolder(path);
```

Comment:
What I'm working to achieve here is the ability have multiple (disparate) folders on the files system serve as public folders (to emulate a feature of the [Apache HTTP Server](https://httpd.apache.org/) that I know and love so well).

These folders may already exist (e.g., they have static assets in them) or they may not exist (e.g., the folder may be created during program execution, and items will be uploaded to the folder after the server is running, to be served out later as requested). In either case, I need the path.

I'd be open to the suggestion of a service or static factory to manage this, but we should put that into another ticket.